src/Dialect/ONNX/Rewrite.td

```diff
@@ -431,6 +439,18 @@ def CastEliminationPattern : Pat<
 //  (ONNXCastOp (ONNXCastOp $arg, $_), $type),
 //  (ONNXCastOp $arg, $type)>;
 
+//===----------------------------------------------------------------------===//
+// Canonicalization for ONNXCastLikeOp
+//===----------------------------------------------------------------------===//
+
+def CastLikeEliminationPattern:  Pat<
+  // Remove a CastLike pattern if the first input tensor and second input tensor are the same. 
+  (ONNXCastLikeOp $val, $castTo, $saturate),
+  // Remove the castlike.
+  (replaceWithValue $val),
+  // Check that the first input tensor and second input tensor are the same
+  [(HaveSameShape $val, $castTo)]>;
```

Comment:
Quick comment, checking shape is not enough to know two inputs are the same. It would be "$val == $castTo"

Reply:
Thanks for the feedback…. I saw an ‘equal’ constraint. I can use that maybe

Reply:
Updated!

Reply:
Wait nvm... I did a check for the same shape and same element type